### PR TITLE
Add C# sample running an augmented model

### PR DESCRIPTION
**Pull request by natke**
Sample for generating an augmented model and running it in C#

### CODE REVIEW

**Comment by wenbingl** [comment]
Nice pictures, is there any copyright issue?

**Review by wenbingl — APPROVED** [review]

**Comment by natke** [comment]
> Nice pictures, is there any copyright issue?

I think it should be fine. The original model is in the zoo, and the augmented one is generated by the sample.

File: c_sharp/image_classification/model/augment_model.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+mobilenet_model = onnx.load_model('mobilenetv2-12.onnx')
+augmented_model = pnp.SequentialProcessingModule(
+    pnp.PreMobileNet(224),
+    mobilenet_model,
+    pnp.PostMobileNet())
+
+# The image size is dynamic, the 400x500 here is to get a fake input to enable export
+fake_image_input = torch.ones(500, 400, 3).to(torch.uint8)
+augmented_model.forward(fake_image_input)
```
Comment (skottmckay) [inline c_sharp/image_classification/model/augment_model.py:13]: This is slightly misleading as it's not channels last, Something like 3, 500, 400 would be clearer.

File: c_sharp/image_classification/model/augment_model.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import onnx
+import torch
+from onnxruntime_extensions import pnp
+
+mobilenet_model = onnx.load_model('mobilenetv2-12.onnx')
+augmented_model = pnp.SequentialProcessingModule(
```
Comment (skottmckay) [inline c_sharp/image_classification/model/augment_model.py:6]: Where does this model come from? I wasn't part of the PR (binary files need to be manually added).

File: c_sharp/image_classification/README.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
+# Image classification using an augmented ONNX model
+
+This samples uses the utilities in [ONNX Runtime extensions](https://github.com/microsoft/onnxruntime-extensions) to augment the MobileNetV2 model to include the data pre and post processing operations required by the model.
```
Comment (skottmckay) [inline c_sharp/image_classification/README.md:3]: >This samples uses the utilities in [ONNX Runtime extensions](https://github.com/microsoft/onnxruntime-extensions) to augment the MobileNetV2 model to include the data  [](http://example.com/codeflow?start=0&length=167)

nit: This sample...

I think this should also call out it's the PyTorch mobilenet (maybe with a link to it) as the pre-processing is specific to how that model was trained. 

TF does not use per-channel normalization so the pre-processing would be incorrect if it was TF mobilenet.

File: c_sharp/image_classification/README.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+A simple C# console application to make predictions with the augmented model is contained in this directory.
+
+## Pre-requisites
+
+* Install cmake
+* Install gcc for your platform
+* Install conda
+* Create a conda environment
+* Install onnxruntime-extension package from source: python -m pip install git+https://github.com/microsoft/onnxruntime-extensions.git
+
```
Comment (skottmckay) [inline c_sharp/image_classification/README.md:16]: Should this list all the other pip modules required, like 'torch'?

File: c_sharp/image_classification/model/augment_model.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    pnp.PreMobileNet(224),
+    mobilenet_model,
+    pnp.PostMobileNet())
+
```
Comment (skottmckay) [inline c_sharp/image_classification/model/augment_model.py:10]: Some comments in the code explaining what each line is doing, or link to doco on the ort-extensions repo, would help a user understand what is going on.
